debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresConnection.java

```diff
@@ -128,6 +138,13 @@ public PostgresConnection(PostgresConnectorConfig config, TypeRegistry typeRegis
             final PostgresValueConverter valueConverter = PostgresValueConverter.of(config, this.getDatabaseCharset(), typeRegistry);
             this.defaultValueConverter = new PostgresDefaultValueConverter(valueConverter, this.getTimestampUtils(), typeRegistry);
         }
+
+        PostgresConnection.FACTORY = factory;
+        this.jdbcConfig = config.getJdbcConfig();
+    }
+
+    public PostgresConnection(PostgresConnectorConfig config, TypeRegistry typeRegistry, String connectionUsage) {
```

Comment:
This might not be required if the FACTORY variable can be set as config.getConnectionFactory

Reply:
The `config` object is from the `JdbcConfiguration` class and we do not have any control over it, if we use `config.getConnectionFactory()` then we will always be getting the connection factory for single host pattern.

Reply:
Can just set the FACTORY instance variable here and everything else can be same as in existing implementation